scim/data_user.go

```diff
@@ -71,6 +75,7 @@ func DataSourceUser() *schema.Resource {
 			d.Set("home", fmt.Sprintf("/Users/%s", user.UserName))
 			d.Set("repos", fmt.Sprintf("/Repos/%s", user.UserName))
 			d.Set("external_id", user.ExternalID)
+			d.Set("application_id", user.ApplicationID)
```

Comment:
Why?

Reply:
if databricks_current_user  is a Service Principal, return the application_id as well